### DISCUSSION TITLE
Initial reconnect best practices

### DISCUSSION THREAD

**Discussion by TannerGabriel** [miguelgrinberg/python-socketio#998] [Q&A] [answered]
Initial reconnection does not seem to be implemented in the library as mentioned in [this old issue](https://github.com/miguelgrinberg/python-socketio/issues/409). Still, I would like to ask for the best practices for handling the reconnection behavior in your application. Currently, I'm checking for a connection error and exiting the application if the connection cannot be established (the application is automatically restarted).

```python
try:
	sio.connect(
		"https://DOMAIN/",
		socketio_path="/socket.io",
		headers={"x-auth-token": TOKEN},
	)
except socketio.exceptions.ConnectionError:
	sys.exit('Initial connection to Socket.IO server failed')
```

Are there any best practices for this behavior, or is there a plan to implement the initial reconnection functionality?

**Comment by miguelgrinberg** [comment, marked as answer]
What I've seen people do is put the connect call in a for or while loop:

```
for retry in range(3):
    try:
        sio.connect(
            "https://DOMAIN/",
            socketio_path="/socket.io",
            headers={"x-auth-token": TOKEN},
        )
    except socketio.exceptions.ConnectionError:
	sleep(5)
    else:
        break
```

I'm not opposed to adding something like this as an option. Will definitely consider it when I look at https://github.com/miguelgrinberg/python-socketio/issues/485, but that one is also lowish priority in my queue.

**Reply by TannerGabriel** [reply]
Thanks, this is a nice workaround for the problem.  Let me know if you decide to implement it. I would be glad to try it out for your.
